src/lib/util.ml

```diff
@@ -82,3 +82,5 @@ let midpoint x y = (x +. y) /. 2.0
 let dx = 2.22044e-16
 
 let significantly_different_from ?(d=dx) x y = y < (x -. d) || y > (x +. d)
+
+let is_nan x = x == nan
```

Comment:
Thank you for adding this, but it's not the right way. For example, it will fail on ` [|nan|].(0)`. The standard way to define it is
```OCaml
let is_nan x = x <> x
```